C/common/reading.cpp

```diff
@@ -225,6 +225,46 @@ ostringstream convert;
 	return convert.str();
 }
 
+/**
+ * Convert time since epoch to a formatted m_timestamp DataTime in UTC 
+ * and use a cache to speed it up
+ * @param tv_sec	Seconds since epoch
+ * @param date_time	Buffer in which to return the formatted timestamp
+ * @param dateFormat	Format: FMT_DEFAULT or FMT_STANDARD
+ */
+void Reading::getFormattedDateTimeStr(const time_t *tv_sec, char *date_time, readingTimeFormat dateFormat) const
+{
+	static unsigned long cached_sec_since_epoch = 0;
+	static char cached_date_time_str[DATE_TIME_BUFFER_LEN] = "";
+
+	if(strlen(cached_date_time_str) && cached_sec_since_epoch && *tv_sec == cached_sec_since_epoch)
```

Comment:
It woudl be quicker to change this to if (*cached_date_time_str rather than if (strlen(cached_date_time_str as the strlen will have to read every character of the string.